include/parallel_initialisation.hpp

```diff
@@ -0,0 +1,214 @@
+#ifndef __NESO_PARTICLES_PARALLEL_INITIALISATION
+#define __NESO_PARTICLES_PARALLEL_INITIALISATION
+
+#include "particle_group.hpp"
+#include <memory>
+
+namespace NESO::Particles {
+
+namespace {
+
+/**
+ * Function used by parallel_advection_initialisation to step particles along a
+ * line to destination positions. Gets a safe temporary position in the local
+ * subdomain from which to start particle movement.
+ *
+ * @param particle_group ParticleGroup being initialised.
+ * @param point Output point in local subdomain.
+ */
+inline void get_point_in_local_domain(ParticleGroupSharedPtr particle_group,
+                                      double *point) {
+
+  auto mesh = particle_group->domain->mesh;
+  bool is_defined = mesh->get_point_in_subdomain(point);
+  NESOASSERT(is_defined,
+             "This mesh type does not define get_point_in_subdomain which is "
+             "required to use this parallel initialisation system.");
+}
+
+/**
+ *  Function used by parallel_advection_initialisation to step particles along a
+ * line to destination positions. Stores the original particle positions.
+ *
+ *  @param particle_group ParticleGroup being initialised.
+ */
+inline void parallel_advection_store(ParticleGroupSharedPtr particle_group) {
+
+  const int space_ncomp = particle_group->position_dat->ncomp;
```

Comment:
Yes `space_ncomp` is the dimension of the space particle positions are in.

Reply:
How big is the refactor of `ncomp` and similar to `ndims` because `ndims` is used elsewhere in NESO-Particles. It'd be nice if there weren't two things referring to the same concept.

Reply:
Replaced `space_ncomp` with `space_ndims`.